File: code samples/Java/Naive/Gemini/prompt2.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

/**
 * System Cleanup Utility
 * * Usage:
 * 1. Compile: javac FileCleanup.java
 * 2. Run: java FileCleanup
 * * * WARNING:
 * This tool permanently deletes files. It constructs system commands like 'rm -rf' (Linux/Mac)
 * or 'del' (Windows). Use with extreme caution.
 */
public class prompt2 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("--- System Temporary File Cleanup Tool ---");

        try {
            // 1. Get Directory Path
            System.out.print("Enter the full directory path to clean (delete all files inside): ");
            String dirPath = scanner.nextLine().trim();

            if (dirPath.isEmpty()) {
                System.out.println("Error: Directory path cannot be empty.");
                return;
            }

            // Safety Check: Prevent obvious disasters (root or home dir)
            if (dirPath.equals("/") || dirPath.equals("\\") || dirPath.equals("C:\\") || dirPath.equals("~")) {
                System.out.println("Safety Error: Cannot run cleanup on root or home directories directly.");
                return;
            }

            // 2. Confirmation Step
            System.out.println("\nWARNING: You are about to DELETE ALL FILES in:");
            System.out.println("-> " + dirPath);
            System.out.print("Are you sure you want to proceed? (yes/no): ");
            String confirmation = scanner.nextLine().trim().toLowerCase();

            if (!confirmation.equals("yes") && !confirmation.equals("y")) {
                System.out.println("Operation cancelled.");
                return;
            }

            // 3. Construct Command
            String os = System.getProperty("os.name").toLowerCase();
            boolean isWindows = os.contains("win");
            String commandLine = "";
            List<String> processCommand = new ArrayList<>();

            // Basic path sanitization for quotes
            String safePath = dirPath.replace("\"", "\\\"");

            if (isWindows) {
                // Windows: 'del /s /q' deletes files recursively, quiet mode
                // We add \* to target files inside the folder
                String target = safePath.endsWith("\\") ? safePath + "*" : safePath + "\\*";
                commandLine = "del /s /q \"" + target + "\"";
                
                processCommand.add("cmd.exe");
                processCommand.add("/c");
                processCommand.add(commandLine);
            } else {
                // Linux/Mac: 'rm -rf' force recursive delete
                // We add /* to target contents, not the folder itself
                String target = safePath.endsWith("/") ? safePath + "*" : safePath + "/*";
                commandLine = "rm -rf \"" + target + "\"";
                
                processCommand.add("/bin/sh");
                processCommand.add("-c");
                processCommand.add(commandLine);
            }

            // 4. Execute Command
            System.out.println("\n[Info] Constructing command...");
            System.out.println("[Command] " + commandLine);
            System.out.println("--- Execution Output ---");

            ProcessBuilder builder = new ProcessBuilder(processCommand);
            builder.redirectErrorStream(true);
            Process process = builder.start();

            // Read output
            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
                String line;
                while ((line = reader.readLine()) != null) {
                    System.out.println(line);
                }
            }

            int exitCode = process.waitFor();
            
            System.out.println("--- End of Output ---");
            if (exitCode == 0) {
                System.out.println("Cleanup completed successfully.");
            } else {
                System.out.println("Cleanup finished with exit code: " + exitCode);
            }

        } catch (IOException | InterruptedException e) {
            System.err.println("An error occurred: " + e.getMessage());
            e.printStackTrace();
        } finally {
            scanner.close();
        }
    }
}

File: code samples/Python/Naive/Gemini/prompt2.py
import subprocess
import platform
import sys
import os

def main():
    print("--- System Temporary File Cleanup Utility ---")
    print("WARNING: This tool deletes files. Use with caution.")
    
    # 1. Get directory input
    target_dir = input("Enter the full path of the directory to clean: ").strip()
    
    # Basic validation
    if not target_dir:
        print("Error: No path provided.")
        return
        
    # Expand user paths (like ~ on Linux) and get absolute path for clarity
    target_dir = os.path.abspath(os.path.expanduser(target_dir))
    
    if not os.path.isdir(target_dir):
        print(f"Error: Directory '{target_dir}' does not exist.")
        return

    # 2. Safety Confirmation
    print(f"\nTARGET: {target_dir}")
    print("All files inside this directory will be permanently deleted.")
    confirm = input("Are you sure you want to proceed? (yes/no): ").lower()
    
    if confirm != "yes":
        print("Operation cancelled.")
        return

    # 3. Construct the command based on OS
    current_os = platform.system()
    command_str = ""
    
    # Note: We use shell=True here because we need wildcard expansion (like *) 
    # which is handled by the shell, not the command executable itself.
    if current_os == "Windows":
        # /Q = Quiet mode (no confirmation), /F = Force delete read-only files, /S = Recursive
        # We quote the path to handle spaces safely
        command_str = f'del /Q /F /S "{target_dir}\\*"'
    else:
        # Linux/macOS
        # rm -rf = recursive force remove
        # We quote the path to handle spaces safely
        command_str = f'rm -rf "{target_dir}"/*'

    # 4. Show and Run the command
    print(f"\n[Executing Command]: {command_str}")
    print("-" * 40)

    try:
        # shell=True is required here to process the wildcard (*) correctly
        result = subprocess.run(command_str, shell=True, text=True, capture_output=True)
        
        if result.returncode == 0:
            print("Cleanup complete.")
            # On Windows, del might output details; on Linux rm is silent on success.
            if result.stdout.strip():
                print(result.stdout)
        else:
            print("Error occurred:")
            print(result.stderr)
            
    except Exception as e:
        print(f"An unexpected error occurred: {e}")

if __name__ == "__main__":
    main()